core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameTest.java

```diff
@@ -0,0 +1,145 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.connections.vfs.provider;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.apache.commons.vfs2.FileName;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileType;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ConnectionFileNameTest {
+
+  @Test
+  public void testPvfsRoot() throws Exception {
+    assertPvsRoot( new ConnectionFileName( null ) );
```

Comment:
what are these test cases supporting in the code `null` and` ""`, `VFSConnectionManagerHelper#getConnectionRootFileName` source code blocks this from happening no?


```
  @NonNull
  public ConnectionFileName getConnectionRootFileName( @NonNull VFSConnectionDetails details ) {

    String connectionName = details.getName();
    if ( StringUtils.isEmpty( connectionName ) ) {
      throw new IllegalArgumentException( "Unnamed connection" );
    }

    return new ConnectionFileName( encodeConnectionName( connectionName ) );
  }
```

Reply:
These are exercised via the subclass `PvfsRootFileObject`, created in `ConnectionFileSystem#createFile(.)`, https://github.com/pentaho/pentaho-kettle/pull/9334/files/760edcfbf4c2a6dcae531e7627ce19e50246be93#diff-d0270228220b533357c03be0e65d621a08cfbb704f12d30f58b33d96d5972250R79.